src/components/TrialBanner.js

```diff
@@ -0,0 +1,182 @@
+      pageTwo && div({ style: { marginTop: '0.5rem', padding: '1rem', border: `1px solid ${colors.blue[0]}`, borderRadius: '0.25rem', backgroundColor: '#f4f4f4' } }, [
+        h(LabeledCheckbox, {
+          checked: termsAgreed === 'true',
+          onChange: v => this.setState({ termsAgreed: v.toString() })
+        }, [span({ style: { marginLeft: '0.5rem' } }, ['I agree to the terms of this Agreement.'])]),
+        div({ style: { flexGrow: 1, marginBottom: '0.5rem' } }),
+        h(LabeledCheckbox, {
+          checked: cloudTermsAgreed === 'true',
+          onChange: v => this.setState({ cloudTermsAgreed: v.toString() })
+        }, [
+          span({ style: { marginLeft: '0.5rem' } }, [
+            'I agree to the Google Cloud Terms of Service.', div({ style: { marginLeft: '1.5rem' } }, [
+              'Google Cloud Terms of Service:',
+              a({
+                style: { textDecoration: 'underline', marginLeft: '0.25rem' },
+                target: 'blank',
+                href: 'https://cloud.google.com/terms/'
+              }, ['https://cloud.google.com/terms/', icon('pop-out', { style: { marginLeft: '0.25rem' } })])
+            ])
+          ])
+        ])
+      ])
+    ])
+  }
+
+  render() {
+    const { authState: { isSignedIn, profile }, ajax: { User } } = _.omit('isVisible', this.props)
+    const { accessingCredits, loading, finalizeTrial } = this.state
+    const { trialState } = profile
+    if (!trialState || !isSignedIn || trialState === 'Finalized') return null
+    const { 'Enabled': { title, message, enabledLink, button, isWarning } } = messages
+
+    return div([
+      div({
+        style: {
+          flex: 1, display: 'flex', alignItems: 'center', padding: '1.5rem', height: 110,
+          backgroundColor: isWarning ? colors.orange[0] : '#359448',
+          justifyContent: 'center', color: 'white', width: '100%', fontSize: '1rem'
+
+        }
+      },
+      [
+        div({
+          style: {
+            fontSize: '1.5rem', fontWeight: 500, textAlign: 'right', borderRight: '1px solid', paddingRight: '1rem', marginRight: '1rem',
+            maxWidth: 200, flexShrink: 0
+          }
+        }, title),
+        span({ style: { maxWidth: 600, lineHeight: '1.5rem' } },
+          [
+            message,
+            enabledLink && a({
+              style: { textDecoration: 'underline', marginLeft: '0.5rem' },
+              target: 'blank',
+              href: enabledLink.url
+            }, [enabledLink.label, icon('pop-out', { style: { marginLeft: '0.25rem' } })])
+          ]),
+        h(Clickable, {
+          style: {
+            display: 'block', fontWeight: 500, fontSize: '1.125rem', border: '2px solid', borderRadius: '0.25rem', padding: '0.5rem 1rem',
+            marginLeft: '0.5rem', flexShrink: 0
+          },
+          onClick: () => {
+            button.isExternal ? window.open(button.url, '_blank') : this.setState({ accessingCredits: true })
```

Comment:
TOL: It's a little weird that this does two completely different things depending on the current status. This feels like two different buttons to me, though I'm not sure it would be straightforward to do that without a structural change.

Reply:
Yeah, the button changes its contents a few times: from starting the trial, to a loading spinner, and finally to the "Learn more". I think this is the cleanest way to write it without splitting it into potentially three different buttons with the same styling, but differing actions that correspond to certain conditions. Splitting them up wouldn't be difficult, but I'm not sure which way would be better.